pkg/types/validation/installconfig.go

```diff
@@ -163,6 +165,21 @@ func ipnetworksToStrings(networks []ipnet.IPNet) []string {
 	return diag
 }
 
+// networkTypeRequiresHostPrefix checks whether the hostPrefix for the networkType needs to be
+// validated or not depending on if the networkType uses the hostPrefix.
+// It returns true if the networkType requires hostPrefix or if the networkType has a non-zero hostPrefix.
+func networkTypeRequiresHostPrefix(networkType string, hostPrefix int32) bool {
+	// list of known plugins that require hostPrefix to be set
+	pluginsUsingHostPrefix := sets.NewString(string(operv1.NetworkTypeOpenShiftSDN), string(operv1.NetworkTypeOVNKubernetes))
+	if pluginsUsingHostPrefix.Has(networkType) {
+		return true
+	}
+	if hostPrefix != 0 {
+		return true
```

Comment:
It's weird to mix these two checks. I'd move the `hostPrefix != 0` check out of this function into `validateClusterNetwork`